pootle/apps/virtualfolder/migrations/0013_set_projects_languages.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+from pootle.core.url_helpers import split_pootle_path
+
+
+def parse_filter_rule(dir_path, filter_rule, lang_from_filters, proj_from_filters):
+    filter_rule = "/%s/" % filter_rule.strip("/")
+    rule_lang = rule_proj = None
+    new_rule = filter_rule
+    if lang_from_filters:
+        rule_lang, rule_proj = split_pootle_path(filter_rule)[:2]
+        new_rule = filter_rule[len(rule_proj) + len(rule_lang) + 2:]
+    elif proj_from_filters:
+        rule_proj = split_pootle_path("/X/%s/" % filter_rule)[1]
+        new_rule = filter_rule[len(rule_proj) + 1:]
+    elif dir_path:
+        new_rule = filter_rule[len(dir_path):]
```

Comment:
Assuming original location was `/af/firefox/dir/` then `dir_path` will be `dir`. If we assume some original rule `subdir/file.po` then `filter_rule` will be `/subdir/file.po/` thus the new rule will be `bdir/file.po/`.